commcare_connect/program/views.py

```diff
@@ -66,3 +72,42 @@ def get_template_names(self):
         view = ("add", "edit")[self.object is not None]
         template = f"program/program_{view}.html"
         return template
+
+
+class ManagedOpportunityList(ProgramManagerMixin, ListView):
+    model = ManagedOpportunity
+    paginate_by = 10
+    default_ordering = "name"
+    template_name = "opportunity/opportunity_list.html"
+
+    def get_queryset(self):
+        ordering = self.request.GET.get("sort", self.default_ordering)
+        ordering = ALLOWED_ORDERINGS.get(ordering, self.default_ordering)
+        program_id = self.kwargs.get("pk")
+        return ManagedOpportunity.objects.filter(program_id=program_id).order_by(ordering)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["program_id"] = self.kwargs.get("pk")
+        context["add_opp_url"] = reverse(
```

Comment:
Rather than introducing a specific context key for managed_opportunity url, use the same key such as `opportunity_init_url` and checking for it in the template [below](https://github.com/dimagi/commcare-connect/pull/377/files#diff-1f71f2b8b52cf2bf5ab2f3ed96a2448172eb702eca858f64e6e423ed3233b4c5R18), override them separately in `ManagedOpportunityList` and `OpportunityList` and use the same key in the template.